Include/README.rst

```diff
@@ -0,0 +1,68 @@
+The Python C API
+================
+
+The C API is divided into three sections:
+
+1. ``Include/``
+2. ``Include/cpython/``
+3. ``Include/internal/``
+
+
+Include: Limited API and Stable ABI
+===================================
+
+``Include/``, excluding the ``cpython`` and ``internal`` subdirectories,
+contains the public Limited API (Application Programming Interface) and
+the Stable ABI (Application Binary Interface).  :pep:`384` defined the
+Stable ABI and the Limited API; two related, but distinct concepts.
```

Comment:
PEP 384 doesn't seem to make a difference between "stable ABI" and "limited API". What makes you think the contrary?

Reply:
Quoting from PEP 384:
```
The ABI specification falls into two parts: an API specification, specifying what function (groups) are available for use with the ABI, and a linkage specification specifying what libraries to link with.
```

Quoting from the draft PEP 652:
```
PEP 384 and this document deal with the Limited API and the Stable ABI, two related but distinct concepts. In short:

- The Stable ABI is a promise that certain extensions compiled with CPython 3.x will be binary compatible with all subsequent versions of CPython 3.x.
- The Limited API is a subset of CPython's C API that produces such extensions.
```

As you can see, I've stolen the wording from PEP 652.

If this is wrong, can you point me in the right direction?

Reply:
Hmm... well, it is misleading because it conveys the impression that the stable ABI and the limited API contain _different_ functions. Actually, they are entirely identical.

If you want to stress the distinction between those concepts, you could say something like:
> The limited API is a subset of C API designed to guarantee ABI stability accross Python versions on any given platform (hence it is sometimes also referred to as "the stable ABI").

Reply:
The Stable ABI does not guarantee binary compatibility across platforms. I think that is impossible :) Anyway, the Limited API and the Stable ABI are indeed intertwined. Maybe we should tone down the Stable ABI in this text:

```diff
-Include: Limited API and Stable ABI
-===================================
+Include: Limited API
+====================
 
 ``Include/``, excluding the ``cpython`` and ``internal`` subdirectories,
-contains the public Limited API (Application Programming Interface) and
-the Stable ABI (Application Binary Interface).  :pep:`384` defined the
-Stable ABI and the Limited API; two related, but distinct concepts.
-The Stable ABI promises binary compatibility across subsequent versions of
-CPython 3.x, given that the source code restricts itself to the Limited
-API.  The Limited API is a subset of the C API.
+contains the public Limited API (Application Programming Interface).
+The Limited API is a subset of the C API, designed to guarantee ABI
+stability across Python 3 versions, and is defined in :pep:`384`.
```

Reply:
They are not really intertwined, they are the same.
Ok with your wording otherwise.

Reply:
According to PEP 652, they are not. I any case, I think I prefer the shorter text that only mentions the limited API.

Reply:
A macro can be in the limited API, but it is not a part of any ABI. An underscored function or variable can be not in the limited API, but be in the stable ABI if it is used in the expansion of a macro from the limited API. We should keep even no longer used private functions if they are in the stable ABI (e.g. `_PyTrash_deposit_object`).